ggu/main.c

```diff
@@ -16,11 +16,380 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <stdbool.h>
+
+#include <rte_ip.h>
+#include <rte_udp.h>
+#include <rte_log.h>
+#include <rte_lcore.h>
+#include <rte_malloc.h>
+#include <rte_ethdev.h>
+#include <rte_atomic.h>
+
+#include "gatekeeper_gk.h"
 #include "gatekeeper_ggu.h"
+#include "gatekeeper_net.h"
+#include "gatekeeper_main.h"
+#include "gatekeeper_config.h"
+
+#define GGU_PD_VER1 (1)
+
+static rte_atomic16_t num_ggu_conf_alloc = RTE_ATOMIC16_INIT(0);
+
+static void
+process_single_policy(struct ggu_policy *policy, struct ggu_config *ggu_conf)
+{
+	/*
+	 * Identify which GK block is responsible for the
+	 * pair <Src, Dst> in the decision.
+	 */
+	int block_id = get_responsible_block(&policy->flow, ggu_conf->gk);
+	struct gk_cmd_entry entry;
```

Comment:
Given that the goal is to send `entry` through a mailbox, why don't you already allocate `entry` here? I don't see the point of having `entry` on stack here, and allocate and copy its content later in `send_gk_cmd()`.